### PR TITLE
WPF: Fix airspace issue to synchronize content over video

### PR DESCRIPTION
**Pull request by DengQingFeng**
### Description of Change ###

Fix the incorrect position of the top element of the video

### Issues Resolved ### 

- fixes #474
[https://code.videolan.org/videolan/LibVLCSharp/-/issues/474](https://code.videolan.org/videolan/LibVLCSharp/-/issues/474)

### Changes ###

Added:
 - event _bckgnd.LayoutUpdated -= Wndhost_LayoutUpdated;
 - event _bckgnd.LayoutUpdated += Wndhost_LayoutUpdated;
 - void Wndhost_LayoutUpdated(object？ sender, EventArgs e)
 
 Removed:
 - event _wndhost.LocationChanged -= Wndhost_LocationChanged;
 - event _wndhost.LocationChanged += Wndhost_LocationChanged;
 - void Wndhost_LocationChanged(object？ sender, EventArgs e)
  
### Platforms Affected ### 

- .net

### CODE REVIEW

**Comment by mfkl** [comment]
Thank you for your PR.

I guess you can remove
```csharp
void Wndhost_LocationChanged(object? sender, EventArgs e)
        {
            var locationFromScreen = _bckgnd.PointToScreen(_zeroPoint);
            var source = PresentationSource.FromVisual(_wndhost);
            var targetPoints = source.CompositionTarget.TransformFromDevice.Transform(locationFromScreen);
            Left = targetPoints.X;
            Top = targetPoints.Y;
        }
```

as well.

Can you please reword your commit like "WPF: Fix airspace issue to synchronize content over video" or something. You may as well include "Fix https://code.videolan.org/videolan/LibVLCSharp/-/issues/474" in the body of the commit message.

In the issue, you mentioned that `Wndhost_SizeChanged` was not going to be needed anymore. If that is the case, feel free to remove that too.

Any idea how we can test this @jeremyVignelles? Thinking maybe with the WPF Forms MediaElement backend...

File: src/LibVLCSharp.WPF/ForegroundWindow.cs [hunk 1]
```diff
@@ -85,7 +85,7 @@ void Background_Loaded(object sender, RoutedEventArgs e)
 
             _wndhost.Closing += Wndhost_Closing;
             _bckgnd.SizeChanged += Wndhost_SizeChanged;
-            _wndhost.LocationChanged += Wndhost_LocationChanged;
```
Comment (jeremyVignelles) [inline src/LibVLCSharp.WPF/ForegroundWindow.cs:88]: Is `Wndhost_LocationChanged` still used then ? Can it be removed ?

Comment (DengQingFeng) [reply src/LibVLCSharp.WPF/ForegroundWindow.cs:88]: I thought Wndhost_LocationChanged is no longer be needed.I'm agree with to remove it. @mfkl

**Comment by mfkl** [comment]
```
2021-07-27T09:04:19.0709075Z   D:\a\1\s\src\LibVLCSharp.WPF\ForegroundWindow.cs(61,38): error CS8622: Nullability of reference types in type of parameter 'sender' of 'void ForegroundWindow.Wndhost_LayoutUpdated(object sender, EventArgs e)' doesn't match the target delegate 'EventHandler' (possibly because of nullability attributes). [D:\a\1\s\src\LibVLCSharp.WPF\LibVLCSharp.WPF_dx1s2pbf_wpftmp.csproj]
2021-07-27T09:04:19.0710371Z   D:\a\1\s\src\LibVLCSharp.WPF\ForegroundWindow.cs(88,38): error CS8622: Nullability of reference types in type of parameter 'sender' of 'void ForegroundWindow.Wndhost_LayoutUpdated(object sender, EventArgs e)' doesn't match the target delegate 'EventHandler' (possibly because of nullability attributes). [D:\a\1\s\src\LibVLCSharp.WPF\LibVLCSharp.WPF_dx1s2pbf_wpftmp.csproj]
```

**Comment by DengQingFeng** [comment]
> ```
> 2021-07-27T09:04:19.0709075Z   D:\a\1\s\src\LibVLCSharp.WPF\ForegroundWindow.cs(61,38): error CS8622: Nullability of reference types in type of parameter 'sender' of 'void ForegroundWindow.Wndhost_LayoutUpdated(object sender, EventArgs e)' doesn't match the target delegate 'EventHandler' (possibly because of nullability attributes). [D:\a\1\s\src\LibVLCSharp.WPF\LibVLCSharp.WPF_dx1s2pbf_wpftmp.csproj]
> 2021-07-27T09:04:19.0710371Z   D:\a\1\s\src\LibVLCSharp.WPF\ForegroundWindow.cs(88,38): error CS8622: Nullability of reference types in type of parameter 'sender' of 'void ForegroundWindow.Wndhost_LayoutUpdated(object sender, EventArgs e)' doesn't match the target delegate 'EventHandler' (possibly because of nullability attributes). [D:\a\1\s\src\LibVLCSharp.WPF\LibVLCSharp.WPF_dx1s2pbf_wpftmp.csproj]
> ```
I found that problem you noticed, it might caused by nullable properties.And I'm already correct the code and re-submit it.

**Comment by mfkl** [comment]
Thanks! So, is `Wndhost_SizeChanged` still necessary?

**Comment by DengQingFeng** [comment]
@mfkl Yes, after I tested it, I found that the Wndhost_SizeChanged event is still needed.


https://user-images.githubusercontent.com/59753422/127277855-412fbe6c-2978-4175-9392-9582b3d0fb90.mp4


https://user-images.githubusercontent.com/59753422/127277865-cccb2252-a0c3-497c-a3a0-6322d3d926ef.mp4

**Review by mfkl — APPROVED** [review]

**Review by jeremyVignelles — APPROVED** [review]
LGTM, thanks for the PR 👍

**Comment by mfkl** [comment]
merged https://github.com/videolan/libvlcsharp/commit/3fe10a53efe852b3c40fdef063c25737be076c51, thank you! Will be available in nightly builds until we do another stable release (probably next week).

**Comment by DengQingFeng** [comment]
Happy to do so :)

**Pull request closed without merging**